tests/foreman/ui/test_computeresource.py

```diff
@@ -18,173 +21,197 @@
 class ComputeResource(UITestCase):
     """Implements Compute Resource tests in UI"""
 
-    @data(*generate_strings_list(len1=8))
-    def test_create_resource_1(self, name):
-        """@Test: Create a new libvirt Compute Resource
+    current_libvirt_url = (LIBVERT_RESOURCE_URL
+                           % conf.properties['main.server.hostname'])
+    current_docker_url = (DOCKER_RESOURCE_URL
+                          % conf.properties['main.server.hostname'])
+
+    @data(*valid_data_list())
+    def test_create_libvirt_resource_different_names(self, name):
+        """@Test: Create a new libvirt Compute Resource using different value
+        types as a name
 
         @Feature: Compute Resource - Create
 
-        @Assert: A libvirt Compute Resource is created
+        @Assert: A libvirt Compute Resource is created successfully
 
         """
-        libvirt_url = "qemu+tcp://%s:16509/system"
-        provider_type = FOREMAN_PROVIDERS['libvirt']
-        url = (libvirt_url % conf.properties['main.server.hostname'])
         with Session(self.browser) as session:
```

Comment:
probably not a good place to ask my question, anyway...
Wouldn't it be better if we initiated a single session in the `setUpClass` method, so we don't need to do it again and again for each tested piece of data?
It would perform the login bit once at startup and the rest would be executed with already existing session - making the tests much faster (especially the UI ones).
Is there a point to initiate a new session with each bit of data?

This is a question for basically all UI tests we currently have.
Thanks for sharing your thoughts.